trackintel/io/file.py

```diff
@@ -195,38 +195,12 @@ def read_triplegs_csv(*args, columns=None, tz=None, index_col=object(), geom_col
     1         1 2015-11-27 12:00:00+00:00 2015-11-27 14:00:00+00:00  LINESTRING (8.56340 47.95600, 8.64560 47.23345...
     """
     columns = {} if columns is None else columns
-
-    # Warning if no 'index_col' parameter is provided
-    if type(index_col) == object:
-        warnings.warn(
-            "Assuming default index as unique identifier. Pass 'index_col=None' as explicit"
-            + "argument to avoid a warning when reading csv files."
-        )
-    elif index_col is not None:
-        kwargs["index_col"] = index_col
-
-    df = pd.read_csv(*args, **kwargs)
-    df = df.rename(columns=columns)
-
-    # construct geom column
-    df[geom_col] = df[geom_col].apply(wkt.loads)
-
-    # transform to datatime
+    df = pd.read_csv(*args, index_col=index_col, **kwargs)
+    df.rename(columns=columns, inplace=True)
     df["started_at"] = pd.to_datetime(df["started_at"])
     df["finished_at"] = pd.to_datetime(df["finished_at"])
-
-    # set timezone if none is recognized
-    for col in ["started_at", "finished_at"]:
-        if not pd.api.types.is_datetime64tz_dtype(df[col]):
-            df[col] = _localize_timestamp(dt_series=df[col], pytz_tzinfo=tz, col_name=col)
-
-    tpls = gpd.GeoDataFrame(df, geometry=geom_col)
-    if crs:
-        tpls.set_crs(crs, inplace=True)
-
-    # assert validity of triplegs
-    tpls.as_triplegs
-    return tpls
+    df[geom_col] = gpd.GeoSeries.from_wkt(df[geom_col])
```

Comment:
I would suggest it, especially because GeoPandas provides this function to use pygeos internally and we get a free speed boost with it. In addition, I also checked `points_from_xy` that function also exists only for GeoPandas &ge; 0.9.0 and we already have that in our code so I think that would be necessary anyway.

Reply:
Could you add this to our requirement file and also anywhere else where applicable?